tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/api/rest/v1/controllers/VariablesController.java

```diff
@@ -52,7 +55,11 @@ public class VariablesController extends ApiErrorController {
   @GetMapping("{variableId}")
   public ResponseEntity<VariableResponse> getVariableById(
       @PathVariable @Parameter(description = "The ID of the variable.", required = true)
-          String variableId) {
+          final String variableId) {
+    if (!tasklistPermissionServices.hasWildcardPermissionToReadProcessInstance()) {
```

Comment:
Following up on [our discussion before](https://github.com/camunda/camunda/pull/36742#discussion_r2279385882), have you verified that checking the process instance permission does not create a problem for migration? For example, I am on 8.7 and have all permissions to work with Tasklist tasks. I migrate to 8.8. Will I lack the permission for this endpoint then?

Reply:
Checking the migration code `tasklist-api:read` flattens out to [these permissions](https://github.com/camunda/camunda/blob/main/migration/identity-migration/src/main/java/io/camunda/migration/identity/config/sm/StaticEntities.java#L254-L261) so there would be an issue here.

I would propose that we add these permissions to the migration app, wdyt? Do we need to validate that with anyone?

Reply:
Checking [here](https://camunda.slack.com/archives/C06UYJMMETZ/p1755778657368469)

Reply:
Outcome is to revert to checking the read user task permission. Will prepare a commit shortly